### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by WaveOfDandelions**
GH: https://WaveOfDandelions.github.io/react_list-of-goods-hooks/

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +17,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_ALPHBT = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
```
Comment (choeqq) [inline src/App.tsx:21]: enums are perfect for that

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +17,103 @@ export const goodsFromServer = [
...
+
+function getVisibleGoods(sortType: string, isReversed: boolean): string[] {
```
Comment (choeqq) [inline src/App.tsx:23]:
```suggestion
function getVisibleGoods(sortType: string, isReversed: boolean) {
```
TS is smart enough to infer returned types

File: src/App.tsx [hunk 3]
```diff
@@ -15,48 +17,103 @@ export const goodsFromServer = [
...
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortType !== SORT_FIELD_LENGTH },
+          )}
+          onClick={(): void => setSortType(SORT_FIELD_LENGTH)}
```
Comment (choeqq) [inline src/App.tsx:80]:
```suggestion
          onClick={()=> setSortType(SORT_FIELD_LENGTH)}
```

File: src/App.tsx [hunk 4]
```diff
@@ -15,48 +17,103 @@ export const goodsFromServer = [
...
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': !isReversed },
+          )}
+          onClick={(): void => setIsReversed(!isReversed)}
```
Comment (choeqq) [inline src/App.tsx:92]:
```suggestion
          onClick={()=> setIsReversed(!isReversed)}
```

File: src/App.tsx [hunk 5]
```diff
@@ -15,48 +17,103 @@ export const goodsFromServer = [
...
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const reset = (): void => {
```
Comment (choeqq) [inline src/App.tsx:50]:
```suggestion
  const reset = ()=> {
```

File: src/App.tsx [hunk 6]
```diff
@@ -15,48 +17,103 @@ export const goodsFromServer = [
...
+    setSortType('');
+    setIsReversed(false);
+  };
```
Comment (choeqq) [inline src/App.tsx:53]: you can also provide function for each button - it will be easier to read

**Review by choeqq — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 5 above)
Comment (WaveOfDandelions) [reply src/App.tsx]: > 

Fixed

File: src/App.tsx (hunk 4 above)
Comment (WaveOfDandelions) [reply src/App.tsx]: > 

Fixed

File: src/App.tsx (hunk 3 above)
Comment (WaveOfDandelions) [reply src/App.tsx]: > 

Fixed

File: src/App.tsx (hunk 2 above)
Comment (WaveOfDandelions) [reply src/App.tsx]: > TS is smart enough to infer returned types

Fixed

File: src/App.tsx (hunk 1 above)
Comment (WaveOfDandelions) [reply src/App.tsx]: > enums are perfect for that

I was add enums

**Review by choeqq — APPROVED** [review]

**Pull request closed without merging**